backend/src/Resolvers.js

```diff
@@ -0,0 +1,32 @@
+module.exports = {
+    Query: {
+        posts: (parent, args, {dataSources}) => dataSources.postsDataStore.allPosts(),
+        users: (parent, args, {dataSources}) => dataSources.userDataStore.allUsers()
+    },
+
+    Post: {
+        author: (parent, args, {dataSources}) => dataSources.userDataStore.getUserById(parent.author_id),
+        votes: async (parent, args, {dataSources}) => {
+            return parent.votes.length;
+        }
+    },
+
+    User: {
+        posts: async (parent, args, {dataSources}) => {
+            const posts = await dataSources.postsDataStore.allPosts();
+            return posts.filter(post => post.author_id === parent.name);
+        }
```

Comment:
```suggestion
        posts: (parent, args, {dataSources}) => {
            const posts = dataSources.postsDataStore.allPosts();
            return posts.filter(post => post.author_id === parent.name);
        }
```
`PostDataStore.allPosts()` is not returning a promise, so why using `async` function here?